music/demos/piano_genie.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Tone from 'tone';
+
+import * as mm from '../src/index';
+
+import {CHECKPOINTS_DIR} from './common';
+
+mm.logging.verbosity = mm.logging.Level.DEBUG;
+
+// tslint:disable-next-line:max-line-length
+const GENIE_CHECKPOINT = `${CHECKPOINTS_DIR}/piano_genie/model/epiano/stp_iq_auto_contour_dt_166006`;
+const LOWEST_PIANO_KEY_MIDI_NOTE = 21;
+
+const genie = new mm.PianoGenie(GENIE_CHECKPOINT);
+
+function initControlsAndAudio () {
+  let lastOutput = -1;
+  let lastTime = new Date();
+  lastTime.setSeconds(lastTime.getSeconds() - 100000);
+  
+  const buttonToNote = new Map<number, number>();
+
+  const synth = new Tone.PolySynth(8, Tone.FMSynth).toMaster();
+
+  // Bind keyboard controls
+  document.onkeydown = (evt: KeyboardEvent) => {
+    if (Tone.context.state !== 'running') {
+      Tone.context.resume();
+    }
+    const key = evt.keyCode;
+    const button = key - 49;
+    if (button >= 0 && button < 8) {
+      if (buttonToNote.has(button)) {
+        return;
+      }
+
+      const time = new Date();
+      const deltaTime = (time.getTime() - lastTime.getTime()) / 1000;
+      const logits = genie.evaluate(button, lastOutput, deltaTime);
```

Comment:
API comment: In general, our models tend to return a softmax or something directly, rather than asking the user to do this. We also definitely don't return any tensors to the users, so I don't think this is the right API here.

What if `genie.evaluate` returned literally the note and nothing else?